addons/account/models/chart_template.py

```diff
@@ -940,7 +940,7 @@ def _parse_csv(self, template_code, model, module=None):
                                     else value
                                 )
                                 for key, value in row.items()
-                                if key != 'id' and value != ""
+                                if key != 'id' and value != "" and ('@' in key or '/' in key or key in model_fields)
```

Comment:
In which case is this failing? Modules not all yet completely loaded? If that is the case, we should delay the update instead I think

If needed, I don't think we should limit it to CSV

Reply:
This line fails: `else (value and ast.literal_eval(value) or False) if model_fields[key].type in ('boolean', 'int', 'float')`
Note it's not a matter waiting for the module to be loaded, because this aims to not fail if the **uninstalled**. So far we've observed the issue (both of them) only for translations load.

We may argue whether we should update anything or not if the module is uninstalled. I opted by allowing the update and just avoid errors when some fields are missing (but I'm looking forward to read better suggesitons ;)). The alternative, i.e skipping non-installed CoAs, may cause issues for generic_coa. At least that was the case with my attempt with something like:
```py
if (get_all or template['visible']) and (include_uninstalled or template['installed'])
```
here https://github.com/odoo/odoo/blob/5eeeb0e3230c065ba8db023ba21ecf7ed572a33b/addons/account/models/chart_template.py#L98

Reply:
> The alternative, i.e skipping non-installed CoAs, may cause issues for generic_coa

Why? It is always installed and installable

Reply:
This is the situation for which the loading fails (bullet 2), on an upgrade `saas-16.2`->`saas-16.3`
```
=> select chart_template,id, partner_id from res_company
+----------------+----+------------+
| chart_template | id | partner_id |
|----------------+----+------------|
| de_skr03       | 1  | 1          |
+----------------+----+------------+
=> select name,id,state, author from ir_module_module where name ilike '%l10n_%' and state='installed';
+-----------------------+-----+-----------+---------------+
| name                  | id  | state     | author        |
|-----------------------+-----+-----------+---------------|
| l10n_din5008          | 216 | installed | Odoo S.A.     |
| l10n_uk_reports       | 134 | installed | SmartMode LTD |
| l10n_din5008_purchase | 877 | installed | Odoo S.A.     |
| l10n_uk               | 669 | installed | SmartMode LTD |
| l10n_din5008_sale     | 181 | installed | Odoo S.A.     |
| l10n_din5008_stock    | 489 | installed | Odoo S.A.     |
+-----------------------+-----+-----------+---------------+
```
I cannot reproduce the issue in standard, no idea how it got there but posting just in case. In principle when we uninstall l10n_de it will uninstall also l10n_de_skr03 and set the chart_template field in res_company to NULL. But this issue may come from previous upgrades.

> Why? It is always installed and installable

I've pushed the patch we'll see if we get the same error in runbot as in https://runbot.odoo.com/runbot/batch/1096917/build/47335490 (older version of the patch).

Reply:
@william-andre can you check this version of the patch please?